tests/Tokenizer/Transformer/TypeAlternationTransformerTest.php

```diff
@@ -388,6 +388,25 @@ public function __construct(
                 ) {}
             }',
         ];
+
+        yield 'parameters by reference' => [
+            [
+                19 => CT::T_TYPE_ALTERNATION,
+                21 => CT::T_TYPE_ALTERNATION,
+                89 => CT::T_TYPE_ALTERNATION,
+                91 => CT::T_TYPE_ALTERNATION,
+            ],
+            '<?php
+                f(FOO|BAR|BAZ&$x);
+                function f1(FOO|BAR|BAZ&$x) {}
+                function f2(FOO&BAR&BAZ&$x) {}
+                f(FOO&BAR|BAZ&$x);
+                f(FOO|BAR&BAZ&$x);
+                fn(FOO&BAR&BAZ&$x) => 0;
+                fn(FOO|BAR|BAZ&$x) => 0;
+                f(FOO&BAR&BAZ&$x);
```

Comment:
Since tokens' indexes are machine friendly, but not human friendly (especially with bigger chunk of code), maybe it would be good to explicitly mark fragments that are related to `$expectedTokens`? This test case is pretty tricky, I believe it could help with understanding. But it's only nice to have cosmetic change.

```suggestion
                f(FOO|BAR|BAZ&$x);
                function f1(FOO|BAR|BAZ&$x) {} // Alternation found
                function f2(FOO&BAR&BAZ&$x) {}
                f(FOO&BAR|BAZ&$x);
                f(FOO|BAR&BAZ&$x);
                fn(FOO&BAR&BAZ&$x) => 0;
                fn(FOO|BAR|BAZ&$x) => 0; // Alternation found
                f(FOO&BAR&BAZ&$x);
```